odoo/addons/openupgrade_records/models/openupgrade_record.py

```diff
@@ -111,3 +114,40 @@ def field_dump(self):
                 dict([(x.name, x.value) for x in record.attribute_ids]))
             data.append(repre)
         return data
+
+    @api.model
+    def list_modules(self):
+        """ Return the set of covered modules """
+        self.env.cr.execute(
+            """SELECT DISTINCT(module) FROM openupgrade_record
+            ORDER BY module""")
+        return [module for module, in self.env.cr.fetchall()]
+
+    @staticmethod
+    def _read_manifest(addon_dir):
+        for manifest_name in MANIFEST_NAMES:
+            if os.access(os.path.join(addon_dir, manifest_name), os.R_OK):
+                with open(os.path.join(addon_dir, manifest_name), 'r') as f:
+                    manifest_string = f.read()
+                    return ast.literal_eval(manifest_string)
+        raise ValidationError('No manifest found in %s' % addon_dir)
+
+    @api.model
+    def get_xml_records(self, module):
+        """ Return all XML records from the given module """
+        addon_dir = get_module_path(module)
+        manifest = self._read_manifest(addon_dir)
+        # The order of the keys are important.
+        # Load files in the same order as in
+        # module/loading.py:load_module_graph
+        files = []
+        for key in ['init_xml', 'update_xml', 'data']:
+            if not manifest.get(key):
+                continue
+            for xml_file in manifest[key]:
+                if not xml_file.lower().endswith('.xml'):
+                    continue
+                parts = xml_file.split('/')
+                with open(os.path.join(addon_dir, *parts), 'r') as xml_handle:
+                    files.append(xml_handle.read())
```

Comment:
```suggestion
                paths.append(os.path.join(addon_dir, *parts))
```
See https://github.com/OCA/server-tools/blob/14.0/upgrade_analysis/models/upgrade_record.py#L170.

Reply:
I still think there is a problem here. You changed the API to return paths instead of XML files. But it looks like this https://github.com/OCA/server-tools/blob/14.0/upgrade_analysis/models/upgrade_analysis.py#L472-L473 gives you local data even if the paths are retrieved from the remote.

Reply:
Then tell what is wrong in _parse_paths method. It should work regardless of what path is passed.

Reply:
It looks like they will be interpreted as local paths, from the instance that runs the analysis (14.0 in this case). They are being passed straigt to etree here: https://github.com/OCA/server-tools/pull/2057. How can that return the remote XML data?

Reply:
So, how we should handle that? Should we put a _parse_paths method in v13 and then in the upgrade_analysis call to this remote `_parse_paths` for the remote paths?

Reply:
You have to reinstate that xml files are passed instead of paths, and then solve the UTF-8 error differently.

Reply:
Would you like to make a PR in server-tools to amend that in a way that also solves the utf-8 error? I am lost with etrees and encodings.

Reply:
I don't really have the time right now. It looks like the problem could be in this line: https://github.com/OCA/server-tools/pull/2057/files#diff-e768c0bad8381e3d5a9cef3f3193890e5438592d5e9da9486c523bb55b19805aL422. I suggest to catch encoding exceptions there and just skip that file. It's only l10n_ma/data/l10n_ma_chart_data.xml that encodes with `<?xml version="1.0" encoding="ISO-8859-15"?>` and I think that's just wrong in itself.

Reply:
Ok, I will do that.

Reply:
I did in local this `files` approach instead of `paths` approach, and the noupdate data was still wrong :S